### PR TITLE
chore!: drop node 16 and 19 support

### PR DESCRIPTION
**Pull request by MichaelSun90**
BREAKING CHANGE: Node.js 16.x and 19.x are no longer supported by `tedious`.

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 19.x]
+        node-version: [18.x, 19.x, 20.x, 21.x]
```
Comment (mShan0) [inline .github/workflows/nodejs.yml:49]: We can drop 19 support since it is no longer supported.

File: appveyor.yml [hunk 2]
```diff
@@ -2,7 +2,6 @@ version: "{build}"
 
 environment:
   matrix:
-    - nodejs_version: "16"
     - nodejs_version: "18"
     - nodejs_version: "19"
```
Comment (mShan0) [inline appveyor.yml:6]: We can drop 19 support since it is no longer supported. Should we add 20 and 21 here?

Comment (arthurschreiber) [reply appveyor.yml]: Yes, let's drop support for 19 and add 20 and 21.

Comment (MichaelSun90) [reply appveyor.yml]: Just push a commit for adding 20, 21 and remove 19 👍 @arthurschreiber , the CI still expecting check for node 16 and 19, is this configurable from one of the yml files or it is a ci setting? 🤔

**Pull request merged**